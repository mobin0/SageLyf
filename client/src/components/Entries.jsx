import React from "react";
import ReactDOM from 'react-dom';
import $ from 'jquery';



class Entries extends React.Component {
  constructor ({entry}) {
    super();
    this.state = {
      entry :entry
    };
    // console.log("Entries", this.state.entry)
    this.setMinutes();
  }

  setMinutes() {
    let minutes = this.state.entry.duration;
    let hours = Math.floor(minutes/60);
    minutes = minutes - hours*60;
    // console.log(minutes)
    this.state.minutes = minutes < 10 ? "0"+minutes : "" + minutes;
    this.state.hours= "0" + hours;
  }

  render () {
    return (
      <div className = "entry-course-items"> 
        <div className = "entry-image-div"> <img className = "entry-image" src = "img/playbutton.png"/></div>
        <div className = "entry-name">{this.state.entry.name}</div>
        <div className = "entry-lecture-length">{this.state.hours + ":" + this.state.minutes}</div>
      </div>
    );
  }
}

export default Entries;