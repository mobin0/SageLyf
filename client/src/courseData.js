var courseData = [{"title":"Team-oriented discrete challenge","sectionNumber":1,"entries":[{"name":"Talk by Dr. Kolby Welch","duration":29,"entryNumber":1},{"name":"Talk by Liana Mayer","duration":55,"entryNumber":2},{"name":"Talk by Ulices Braun","duration":16,"entryNumber":3},{"name":"Talk by Raul Klein","duration":66,"entryNumber":4},{"name":"Talk by Ana Hudson","duration":33,"entryNumber":5},{"name":"Talk by Reva Treutel","duration":6,"entryNumber":6},{"name":"Talk by Rigoberto Dietrich II","duration":23,"entryNumber":7},{"name":"Talk by Vicky Daugherty","duration":74,"entryNumber":8}]},{"title":"Reverse-engineered attitude-oriented service-desk","sectionNumber":2,"entries":[{"name":"Talk by Katlynn Davis","duration":52,"entryNumber":1},{"name":"Talk by Mr. Lavern Weissnat","duration":84,"entryNumber":2},{"name":"Talk by Miss Ila Ondricka","duration":4,"entryNumber":3}]}]

export default courseData;