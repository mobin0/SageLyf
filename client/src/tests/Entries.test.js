import React from 'react';
import Entries from "./../components/Entries.jsx";
import renderer from "react-test-renderer"
import { cleanup, fireEvent, waitForElement } from 'react-testing-library';
import {shallow, mount, render, configure} from 'enzyme';
import 'jest-dom/extend-expect';
import Adapter from 'enzyme-adapter-react-16';
import courseData from './../../../db/test.js'


configure({ adapter: new Adapter() });

afterEach(cleanup);

const testCase = courseData[0].entries[0]

describe('Entries', () => {
  // setupSandbox()
  const entries = shallow(<Entries entry = {testCase}/>)

  it ('should have same snapshot', () => {
  expect(entries).toMatchSnapshot()
  })

  it ('should load', () => {
 	expect(entries.exists()).toBeTruthy()
  })

  it ('should have entry-name div', () => {
 	expect(entries.find('.entry-name').length).toBe(1)
  })

  it ('should have entry-name text', () => {
  	expect(entries.find('.entry-name').text()).toBe(testCase.name)
  })

  it ('should have the correct hours', () => {
  	expect(entries.state().hours).toBe("04")
  })

  it ('should have the correct minutes', () => {
  	expect(entries.state().minutes).toBe("09")
  })

  it ('should have the correct time', () => {
 	expect(entries.find('.entry-lecture-length').text()).toBe('04:09')
  })

})
  // const entryName = sel('entry-name')
  // console.log(entryName.text())
//   it('should work', () => {
//     const w = mount(<Greeting />)
//     helloBtn(w).simulate('click')
//   })

//     it('should work', () => {
//     const w = mount(<Greeting />)
//     helloBtn(w).simulate('click')
//   })


//       it('should work', () => {
//     const w = mount(<Greeting />)
//     helloBtn(w).simulate('click')
//   })


//         it('should work', () => {
//     const w = mount(<Greeting />)
//     helloBtn(w).simulate('click')
//   })

//   it('should provide the greeting', async () => {
//     const w = mount(<Greeting />)

//     await reflush(w)

//     expect(w.text()).toEqual('Hello worldGreet me')

//     findByText(/Greet/, w).simulate('click')
//   })
// })


// let entry = {"name":"Talk by Marlen Daniel","duration":196,"entryNumber":1}
//
 // describe('I am a KING', () => {
//   test('works', () => {
//     renderer.create(<Entries entry = {entry}/>)

//   })
// })



// describe('CourseHeader Component', () => {
//   test('renders to the document', () => {
//     render(<CourseHeader />);
//   });

//   test('displays the course title', () => {
//     const { container } = render(<CourseHeader courseData={mockCourseData}/>);
//     expect(container).toHaveTextContent(mockCourseData.title);
//   });

//   test('displays the course subtitle', () => {
//     const { container } = render(<CourseHeader courseData={mockCourseData}/>);
//     expect(container).toHaveTextContent(mockCourseData.subtitle);
//   });

//   test('renders the rating component', () => {
//     const { container } = render(<CourseHeader courseData={mockCourseData} />);
//     const ratingComponent = container.querySelector('.ui.star.rating');
//     expect(ratingComponent).toBeInTheDocument();
//   })

//   test('Renders the rating component with the corrct star count', () => {
//     const { container } = render(<CourseHeader courseData={mockCourseData} />);
//     const starCount = container.querySelectorAll('.active').length;
//     expect(starCount).toBe(correctText.star_count);
//   })


// // describe('My work', () => {
// //   test('works', () => {
// //     const r = renderer.create()
// //   })
// // })
