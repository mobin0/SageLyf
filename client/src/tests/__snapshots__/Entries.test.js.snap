// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Entries
    entry={
      Object {
        "duration": 249,
        "entryNumber": 1,
        "name": "Talk by Agustin Murray",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="entry-image-div"
        >
           
          <img
            className="entry-image"
            src="img/playbutton.png"
          />
        </div>,
        <div
          className="entry-name"
        >
          Talk by Agustin Murray
        </div>,
        <div
          className="entry-lecture-length"
        >
          04:09
        </div>,
      ],
      "className": "entry-course-items",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <img
              className="entry-image"
              src="img/playbutton.png"
            />,
          ],
          "className": "entry-image-div",
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "entry-image",
              "src": "img/playbutton.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Talk by Agustin Murray",
          "className": "entry-name",
        },
        "ref": null,
        "rendered": "Talk by Agustin Murray",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "04:09",
          "className": "entry-lecture-length",
        },
        "ref": null,
        "rendered": "04:09",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="entry-image-div"
          >
             
            <img
              className="entry-image"
              src="img/playbutton.png"
            />
          </div>,
          <div
            className="entry-name"
          >
            Talk by Agustin Murray
          </div>,
          <div
            className="entry-lecture-length"
          >
            04:09
          </div>,
        ],
        "className": "entry-course-items",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <img
                className="entry-image"
                src="img/playbutton.png"
              />,
            ],
            "className": "entry-image-div",
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "entry-image",
                "src": "img/playbutton.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Talk by Agustin Murray",
            "className": "entry-name",
          },
          "ref": null,
          "rendered": "Talk by Agustin Murray",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "04:09",
            "className": "entry-lecture-length",
          },
          "ref": null,
          "rendered": "04:09",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
